zrml/prediction-markets/src/lib.rs

```diff
@@ -1644,6 +1766,16 @@ mod pallet {
             Ok(total_weight)
         }
 
+        fn on_market_open(
+            market_id: &MarketIdOf<T>,
+            market: Market<T::AccountId, T::BlockNumber, MomentOf<T>>,
+        ) -> Result<Weight, DispatchError> {
+            match market.status {
+                MarketStatus::Active => Self::open_market(market_id),
+                _ => Err(Error::<T>::InvalidMarketStatus.into()), // Should never occur!
+            }
+        }
+
```

Comment:
Why are `on_market_open` and `open_market` two separate functions? All that `on_market_open` does is check if it is a valid state transition and call `open_market`, but `open_market` should actually also adhere to the state transition rules, shouldn't it?
Is the idea to adhere to the current conventions?

Reply:
Yes, other than conformity, I have no good reason for keeping it this way. I have zero conviction about keeping this in, so if you think it's better to remove this layer of abstraction, I can go ahead and get rid of it.

Reply:
I would merge them, because using `open_market` on a market in an invalid state should not be possible. I also wonder why `open_market` does not actually open the market, i.e. make it active (in case of a permissionless market). Where is the state changed to `MarketStatus::Active`?

Reply:
Good question. There is no new `MarketStatus`. Even when the market is not yet open, it is active. The only thing that changes when the market opens is that status of the pool.

If you like, I can add an additional variant to `MarketStatus`. The reason that I chose _not_ to do this was that this would be purely informational. There is no technical need for this status in `prediction-markets`.

Reply:
Okay, let's not introduce the state.

Reply:
`on_market_open` removed in ebbe90685bb241c8d7a1e7d882f4552ae5766d59